ocs_ci/deployment/deployment.py

```diff
@@ -678,3 +696,149 @@ def create_catalog_source(image=None, ignore_upgrade=False):
+        mode='w+', prefix='local_storage_operator', delete=False
+    )
+    templating.dump_data_to_temp_yaml(
+        lso_data, lso_data_yaml.name
+    )
+    logger.info("Creating local-storage-operator")
+    run_cmd(f"oc create -f {lso_data_yaml.name}")
+
+    local_storage_operator = ocp.OCP(
+        kind=constants.POD, namespace='local-storage'
+    )
+    assert local_storage_operator.wait_for_resource(
+        condition=constants.STATUS_RUNNING,
+        selector=constants.LOCAL_STORAGE_OPERATOR_LABEL,
+        timeout=600
+    ), "Local storage operator did not reach running phase"
+
+    # Retrieve NVME device path ID for each worker node
+    device_paths = get_device_paths(worker_names)
+
+    # Pull local volume yaml data
+    logger.info("Pulling LocalVolume CR data from yaml")
+    lv_data = templating.load_yaml(
+        constants.LOCAL_VOLUME_YAML
+    )
+
+    # Update local volume data with NVME IDs
+    logger.info(
+        "Updating LocalVolume CR data with device paths: %s",
+        device_paths
+    )
+    lv_data['spec']['storageClassDevices'][0][
+        'devicePaths'
+    ] = device_paths
+
+    # Create temp yaml file and create local volume
+    lv_data_yaml = tempfile.NamedTemporaryFile(
+        mode='w+', prefix='local_volume', delete=False
+    )
+    templating.dump_data_to_temp_yaml(
+        lv_data, lv_data_yaml.name
+    )
+    logger.info("Creating LocalVolume CR")
+    run_cmd(f"oc create -f {lv_data_yaml.name}")
+    logger.info("Waiting 30 seconds for PVs to create")
+    verify_pvs_created(len(worker_names))
+
+
+@retry(AssertionError, 12, 10, 1)
+def verify_pvs_created(num_workers):
+    """
+    Verify that PVs were created and are in the Available state
+
+    Args:
+        num_workers (int): number of worker nodes in the cluster
+
+    Raises:
+        AssertionError: if any PVs are not in the Available state or if the
+            number of PVs does not match the number of workers in the cluster.
+
+    """
+    logger.info("Verifying PVs are created")
+    out = run_cmd("oc get pv -o json")
+    pv_json = json.loads(out)
+    for pv in pv_json['items']:
+        pv_state = pv['status']['phase']
+        pv_name = pv['metadata']['name']
+        logger.info("%s is %s", pv_name, pv_state)
+        assert pv_state == 'Available', (
+            f"{pv_name} not in 'Available' state. Current state is {pv_state}"
+        )
+    num_pvs = len(pv_json['items'])
+    logger.debug("PVs, Workers: %s, %s", num_pvs, num_workers)
+
+
+def get_device_paths(worker_names):
+    """
+    Retrieve a list of the device paths for each worker node
+
+    Args:
+        worker_names (list): worker node names
+
+    Returns:
+        list: device path ids
+    """
+    device_paths = []
+    platform = config.ENV_DATA.get('platform').lower()
+    if platform == 'aws':
+        pattern = 'nvme-Amazon_EC2_NVMe_Instance_Storage'
+    # TODO: add patterns for vsphere and bare metal
```

Comment:
One suggestion which can be taken up in follow up PR:
we can use one or all of the free device available and not just disk matching the pattern having `nvme-..`
In this way we will be able to do LSO based deployment even if we don't have nvme disks in that setup.

Reply:
Do you have a reliable (and automatable) way to determine which disks are free and which are not?